middleware/index.ts

```diff
@@ -37,77 +78,24 @@ function createLogger(requestID?: string): Logger {
     return logger;
 }
 
-function errorMessage(error: Error): any {
+function buildErrorResponse(extraInfo: ExtraInfo, err: Error): any {
+    // any error in user-function-space should be considered a 500 toerhwi
+    // ensure we send down application/json in this case
+    extraInfo.setStack(err.stack);
+    return buildResponse(err instanceof MiddlewareError ? err.code : INTERNAL_SERVER_ERROR_CODE, err.message, extraInfo);
+}
+
+function buildResponse(code: string, response: any, extraInfo: ExtraInfo): any {
     // any error in user-function-space should be considered a 500
     // ensure we send down application/json in this case
     return Message.builder()
         .addHeader('content-type', 'application/json')
-        .addHeader('x-http-status', '500')
-        .payload({
-            error: error.toString(),
-        })
+        .addHeader('x-http-status', code)
+        .addHeader('x-extra-info', JSON.stringify(extraInfo))
```

Comment:
I think the URI-encoding should happen here after JSON.stringify-ing.  That way *all* the special JSON chars get escaped in the header to survive transport.